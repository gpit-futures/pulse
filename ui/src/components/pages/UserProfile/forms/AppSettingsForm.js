import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
