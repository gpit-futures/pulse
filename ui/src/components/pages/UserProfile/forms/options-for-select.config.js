const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
