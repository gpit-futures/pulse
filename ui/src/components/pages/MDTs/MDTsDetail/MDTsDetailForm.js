import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { valuesNames, valuesLabels } from '../forms.config';
import { validateForm } from '../forms.validation';

@reduxForm({
  form: 'mdtsPanelFormSelector',
  validate: validateForm,
})
export default class MDTsDetailForm extends PureComponent {
  componentDidMount() {
    const { detail, initialize } = this.props;
    initialize(this.defaultValuesForm(detail));
  }
  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesNames.TEAM]: value[valuesNames.TEAM],
      [valuesNames.LINK]: value[valuesNames.LINK],
      [valuesNames.QUESTION]: value[valuesNames.QUESTION],
      [valuesNames.NOTES]: value[valuesNames.NOTES],
      [valuesNames.DATE_OF_REQUEST]: value[valuesNames.DATE_OF_REQUEST],
      [valuesNames.DATE_OF_MEETING]: value[valuesNames.DATE_OF_MEETING],
    };

    return defaultFormValues;
  }
  render() {
    const { isSubmit } = this.props;
    return (
      <div className="panel-body-inner">
        <form name="mdtsPanelForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.TEAM}
                  name={valuesNames.TEAM}
                  id={valuesNames.TEAM}
                  type="text"
                  placeholder=""
                  component={ValidatedInput}
                  props={{ isSubmit }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.DATE_OF_REQUEST}
                  name={valuesNames.DATE_OF_REQUEST}
                  id={valuesNames.DATE_OF_REQUEST}
                  component={DateInput}
                  props={{ format: 'DD-MMM-YYYY', isSubmit }}
                />
              </div>
            </div>

            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.DATE_OF_MEETING}
                  name={valuesNames.DATE_OF_MEETING}
                  id={valuesNames.DATE_OF_MEETING}
                  component={DateInput}
                  props={{ format: 'DD-MMM-YYYY', isSubmit }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.LINK}
                  name={valuesNames.LINK}
                  id={valuesNames.LINK}
                  type="text"
                  placeholder=""
                  component={ValidatedInput}
                  props={{ isSubmit }}
                />
              </div>
            </div>

            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.QUESTION}
                  name={valuesNames.QUESTION}
                  id={valuesNames.QUESTION}
                  component={ValidatedTextareaFormGroup}
                  props={{ isSubmit }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.NOTES}
                  name={valuesNames.NOTES}
                  id={valuesNames.NOTES}
                  component={ValidatedTextareaFormGroup}
                  props={{ isSubmit }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}
