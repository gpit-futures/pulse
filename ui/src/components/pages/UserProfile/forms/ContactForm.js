import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
