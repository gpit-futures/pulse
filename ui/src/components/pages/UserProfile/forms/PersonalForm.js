import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
