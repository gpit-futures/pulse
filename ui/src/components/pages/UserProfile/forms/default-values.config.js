import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
