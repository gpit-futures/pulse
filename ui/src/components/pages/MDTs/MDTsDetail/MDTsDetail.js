import React, { PureComponent } from 'react';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel'
import MDTsDetailForm from './MDTsDetailForm'
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels } from '../forms.config';

const MDTS_PANEL = 'mdtsPanel';

export default class MDTsDetail extends PureComponent {
  render() {
    const { onExpand, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, mdtFormValues, isSubmit } = this.props;
    let { detail } = this.props;
    detail = detail || {};
    const dateOfRequest = getDDMMMYYYY(detail[valuesNames.DATE_OF_REQUEST]);
    const dateOfMeeting = getDDMMMYYYY(detail[valuesNames.DATE_OF_MEETING]);

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === MDTS_PANEL || expandedPanel === 'all') && !editedPanel[MDTS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={MDTS_PANEL}
            title="MDT Meeting"
            isOpen={openedPanel === MDTS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={mdtFormValues}
            isBtnShowPanel={false}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.TEAM}</label>
                        <div className="form-control-static">
                          {!detail[valuesNames.TEAM]
                            ? <span>{valuesLabels.TEAM_NOT}</span>
                            : <ul className="list-reset">
                              <li>{detail[valuesNames.TEAM]}</li>
                            </ul>
                          }
                        </div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.DATE_OF_REQUEST}</label>
                        <div className="form-control-static">
                          {!detail[valuesNames.DATE_OF_REQUEST]
                            ? <span>{valuesLabels.DATE_OF_REQUEST_NOT}</span>
                            : <ul className="list-reset">
                              <li>{dateOfRequest}</li>
                            </ul>
                          }
                        </div>
                      </div>
                    </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.DATE_OF_MEETING}</label>
                        {!detail[valuesNames.DATE_OF_MEETING]
                          ? <span>{valuesLabels.DATE_OF_MEETING_NOT}</span>
                          : <ul className="list-reset">
                            <li>{dateOfMeeting}</li>
                          </ul>
                        }
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.LINK}</label>
                        <div className="form-control-static">
                          {!detail[valuesNames.LINK]
                            ? <span>{valuesLabels.LINK_NOT}</span>
                            : <a href={detail[valuesNames.LINK]}>{detail[valuesNames.LINK]}</a>
                          }
                        </div>
                      </div>
                    </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.QUESTION}</label>
                        {!detail[valuesNames.QUESTION]
                          ? <span>{valuesLabels.QUESTION_NOT}</span>
                          : <ul className="list-reset">
                            <li>{detail[valuesNames.QUESTION]}</li>
                          </ul>
                        }
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.NOTES}</label>
                        {!detail[valuesNames.NOTES]
                          ? <span>{valuesLabels.NOTES_NOT}</span>
                          : <ul className="list-reset">
                            <li>{detail[valuesNames.NOTES]}</li>
                          </ul>
                        }
                      </div>
                    </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.SOURCE}</label>
                        <div className="form-control-static">{detail[valuesNames.SOURCE]}</div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}
          {(expandedPanel === MDTS_PANEL || expandedPanel === 'all') && editedPanel[MDTS_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={MDTS_PANEL}
            title="Edit MDT"
            isOpen={openedPanel === MDTS_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={mdtFormValues}
            isBtnShowPanel={false}
          >
            <MDTsDetailForm
              detail={detail}
              isSubmit={isSubmit}
            />
          </PluginDetailPanel> : null}
        </div>
      </div>
    )
  }
}
