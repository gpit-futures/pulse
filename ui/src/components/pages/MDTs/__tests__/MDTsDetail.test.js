import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import MDTsDetail from '../MDTsDetail/MDTsDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';

Enzyme.configure({ adapter: new Adapter() });

const propsForMDTsPanel = {
  detail: {
    [valuesNames.TEAM]: 'MDT Prostate Cancer team',
    [valuesNames.DATE_OF_REQUEST]: 1482170593395,
    [valuesNames.DATE_OF_MEETING]: 1456258262000,
    [valuesNames.TIME_OF_MEETING]: 72662000,
    [valuesNames.LINK]: 'test',
    [valuesNames.QUESTION]: 'Increasing back pain',
    [valuesNames.NOTES]: 'Investigations normal. Review in 3 weeks',
    [valuesNames.SOURCE]: 'ethercis',
    [valuesNames.SOURCE_ID]: '94133578-f505-4e76-b4ed-762462508801',
  },
};

const MDTS_PANEL = 'mdtsPanel';
const CONVERT_DATE_OF_REQUEST = getDDMMMYYYY(propsForMDTsPanel.detail[valuesNames.DATE_OF_REQUEST]);
const CONVERT_DATE_OF_MEETING = getDDMMMYYYY(propsForMDTsPanel.detail[valuesNames.DATE_OF_MEETING]);

describe('Component <MDTsDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<MDTsDetail />);

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: propsForMDTsPanel.detail, expandedPanel: 'all', editedPanel: { [MDTS_PANEL]: false } });
    expect(component.props().className).toEqual('section-detail');
    expect(component.find('PluginDetailPanel')).toHaveLength(1);

    // Testing mdtsPanel
    expect(component.find('PluginDetailPanel').at(0).props().name).toEqual(MDTS_PANEL);
    expect(component.find('PluginDetailPanel').at(0).props().title).toEqual('MDT Meeting');
    expect(component.find('PluginDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().editedPanel).toEqual({ mdtsPanel: false });
    expect(component.find('PluginDetailPanel').at(0).props().isShowControlPanel).toEqual(true);

    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.TEAM);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.DATE_OF_REQUEST);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.DATE_OF_MEETING);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.LINK);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.QUESTION);
    expect(component.find('.control-label').at(5).text()).toEqual(valuesLabels.NOTES);
    expect(component.find('.control-label').at(6).text()).toEqual(valuesLabels.SOURCE);

    expect(component.find('.form-control-static').at(0).text()).toEqual(propsForMDTsPanel.detail[valuesNames.TEAM]);
    expect(component.find('.form-control-static').at(1).text()).toEqual(CONVERT_DATE_OF_REQUEST);
    expect(component.find('.form-control-static').at(2).text()).toEqual(propsForMDTsPanel.detail[valuesNames.LINK]);
    expect(component.find('.form-control-static').at(3).text()).toEqual(propsForMDTsPanel.detail[valuesNames.SOURCE]);


    expect(component.find('.list-reset').at(2).text()).toEqual(CONVERT_DATE_OF_MEETING);
    expect(component.find('.list-reset').at(3).text()).toEqual(propsForMDTsPanel.detail[valuesNames.QUESTION]);
    expect(component.find('.list-reset').at(4).text()).toEqual(propsForMDTsPanel.detail[valuesNames.NOTES]);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <MDTsDetail />);
    // Testing component when detail empty object, expandedPanel is mdtsPanel
    component.setProps({ detail: { [valuesNames.SOURCE_ID]: '94133578-f505-4e76-b4ed-762462508801' }, expandedPanel: MDTS_PANEL, editedPanel: { [MDTS_PANEL]: false } });
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(MDTS_PANEL);

    expect(component.find('.form-control-static').at(0).text()).toEqual(valuesLabels.TEAM_NOT);
    expect(component.find('.form-control-static').at(1).text()).toEqual(valuesLabels.DATE_OF_REQUEST_NOT);
    expect(component.find('.form-control-static').at(2).text()).toEqual(valuesLabels.LINK_NOT);
    expect(component.find('.form-control-static').at(3).text()).toEqual('');

    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel is edited
    component.setProps({ detail: propsForMDTsPanel.detail, expandedPanel: MDTS_PANEL, editedPanel: { [MDTS_PANEL]: true } });
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(MDTS_PANEL);
    expect(component).toMatchSnapshot();
  });
});

