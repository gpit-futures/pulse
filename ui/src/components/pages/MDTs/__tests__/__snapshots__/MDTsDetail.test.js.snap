// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <MDTsDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "mdtsPanel": false,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="mdtsPanel"
      title="MDT Meeting"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Service / Team
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <span>
                      No Multidisciplinary Team
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date of Request
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <span>
                      No MDT date
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date of Meeting
                  </label>
                  <span>
                    Not Available
                  </span>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Link to MDT Web Service Directory
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <span>
                      No Link
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Question For MDT
                  </label>
                  <span>
                    No Question For MDT
                  </span>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Meeting Discussion
                  </label>
                  <span>
                    No Meeting Discussion
                  </span>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <MDTsDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "mdtsPanel": true,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="mdtsPanel"
      title="Edit MDT"
    >
      <ReduxForm
        detail={
          Object {
            "dateOfMeeting": 1456258262000,
            "dateOfRequest": 1482170593395,
            "notes": "Investigations normal. Review in 3 weeks",
            "question": "Increasing back pain",
            "servicePageLink": "test",
            "serviceTeam": "MDT Prostate Cancer team",
            "source": "ethercis",
            "sourceId": "94133578-f505-4e76-b4ed-762462508801",
            "timeOfMeeting": 72662000,
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <MDTsDetail /> should renders with props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "mdtsPanel": false,
        }
      }
      isBtnShowPanel={false}
      isOpen={false}
      isShowControlPanel={true}
      name="mdtsPanel"
      title="MDT Meeting"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Service / Team
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <ul
                      className="list-reset"
                    >
                      <li>
                        MDT Prostate Cancer team
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date of Request
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <ul
                      className="list-reset"
                    >
                      <li>
                        19-Dec-2016
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date of Meeting
                  </label>
                  <ul
                    className="list-reset"
                  >
                    <li>
                      23-Feb-2016
                    </li>
                  </ul>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Link to MDT Web Service Directory
                  </label>
                  <div
                    className="form-control-static"
                  >
                    <a
                      href="test"
                    >
                      test
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Question For MDT
                  </label>
                  <ul
                    className="list-reset"
                  >
                    <li>
                      Increasing back pain
                    </li>
                  </ul>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Meeting Discussion
                  </label>
                  <ul
                    className="list-reset"
                  >
                    <li>
                      Investigations normal. Review in 3 weeks
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  >
                    ethercis
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
